src/server/server_family.h

```diff
@@ -358,6 +361,12 @@ class ServerFamily {
   std::unique_ptr<util::fb2::FiberQueueThreadPool> fq_threadpool_;
   std::shared_ptr<detail::SnapshotStorage> snapshot_storage_;
 
+  std::atomic<bool> client_pause_ = false;
```

Comment:
maybe use class ExecutionState ,  calling this member pause_execution_state
than you can call cancel from the unpause and shutdown flow
and check IsCanceled in is_pause_in_progress

Reply:
I know and I avoided it on purpose. I don't think ExecutionState simplifies anything here. Sure, it has this nice interface of `IsCancelled` but when I just need a flag to use on a condition variable I rather do that since I don't need any of the rest of the functionality of the ExecutionState (submitting an error and applying the handler on a separate fiber). I rather have a flag than a wrapper for simple cases like this.,

Reply:
+1 for the rename